perun-core/src/main/java/cz/metacentrum/perun/core/entry/GroupsManagerEntry.java

```diff
@@ -1458,4 +1458,39 @@ public boolean canExtendMembershipInGroupWithReason(PerunSession sess, Member me
 
 		return getGroupsManagerBl().getIndirectMembershipPaths(sess, member, group);
 	}
+
+	@Override
+	public List<RichMember> getGroupRichMembersByIds(PerunSession sess, int groupId, List<Integer> memberIds, List<String> attrNames) throws GroupNotExistsException, PrivilegeException, AttributeNotExistsException  {
+		Utils.checkPerunSession(sess);
+
+		Group group = groupsManagerBl.getGroupById(sess, groupId);
+		getGroupsManagerBl().checkGroupExists(sess, group);
+
+		// Authorization
+		if ( !AuthzResolver.authorizedInternal(sess, "getGroupRichMembersByIds_int_List<Integer>_List<String>_policy", group)) {
+			throw new PrivilegeException(sess, "getGroupRichMembersByIds");
+		}
+
+		List<AttributeDefinition> attributeDefinitions = new ArrayList<>();
+		for (String attrName : attrNames) {
+			attributeDefinitions.add(getPerunBl().getAttributesManagerBl().getAttributeDefinition(sess, attrName));
+		}
+
+		List<Member> members = perunBl.getMembersManagerBl().getMembersByIds(sess, memberIds);
+		members.removeIf(member -> !AuthzResolver.authorizedInternal(sess, "filter-getGroupRichMembersByIds_int_List<Integer>_List<String>_policy", member));
+		members.removeIf(member -> {
+			try {
+				return !isGroupMember(sess, group, member);
+			} catch (PrivilegeException e) {
+				// if can't check membership, remove from requested members
+				return true;
+			} catch (MemberNotExistsException | GroupNotExistsException e) {
```

Comment:
These catch blocks can be collapsed into a single one.
```suggestion
			} catch (MemberNotExistsException | GroupNotExistsException | PrivilegeException e) {
```